staramr/blast/plasmidfinder/PlasmidfinderBlastDatabase.py

```diff
@@ -12,12 +14,54 @@
 
 class PlasmidfinderBlastDatabase(AbstractBlastDatabase):
 
-    def __init__(self, database_dir: str) -> None:
+    def __init__(self, database_dir: str, bacteria=None) -> None:
         """
         Creates a new PlasmidfinderBlastDatabase.
-        :param database_dir: The specific ResFinder database (drug class) directory.
+        :param database_dir: The specific PlasmidFinder database directory.
         """
+
         super().__init__(database_dir)
+        self.plasmidfinder_database_dir = path.join(self.database_dir)
+        self.bacteria = bacteria
+
+        if bacteria is not None:
+            self.plasmidfinder_database_file = path.join(self.database_dir, bacteria)
+            self.plasmidfinder_database_file += ".fsa"
+
+            if (not path.isfile(self.plasmidfinder_database_file)):
+                raise Exception(
+                    "Error, plasmidfinder bacteria [" + bacteria + "] is either incorrect or plasmidfinder database not installed properly")
+            elif bacteria not in PlasmidfinderBlastDatabase.get_bacterias(database_dir):
+                raise Exception("Plasmidfinder bacteria [" + bacteria + "] is not valid")
 
     def get_name(self) -> str:
         return 'plasmidfinder'
+
+    def get_database_names(self):
+        if self.bacteria is not None:
+            names = [self.bacteria]
+        else:
+            names = [f[:-len(self.fasta_suffix)] for f in listdir(self.plasmidfinder_database_dir) if
```

Comment:
If this code is identical to the superclass method `get_database_names()` then you can call it with:

```python
names = super().get_database_names()
```

Reply:
Resolved in dbfcea1